diff --git a/src/xdisp.c b/src/xdisp.c
index a9eb47720d0..deebfeb3c39 100644
--- a/src/xdisp.c
+++ b/src/xdisp.c
@@ -23338,7 +23338,11 @@ append_space_for_newline (struct it *it, bool default_face_p)
 		}
 	      if (extra_line_spacing > 0)
 		{
-		  it->descent += extra_line_spacing;
+		  // TRUNG: Vertically centralize extra line spacing
+		  int extra_line_spacing_ascent = extra_line_spacing / 2;
+		  int extra_line_spacing_descent = extra_line_spacing - extra_line_spacing_ascent;
+		  it->ascent += extra_line_spacing_ascent;
+		  it->descent += extra_line_spacing_descent;
 		  if (extra_line_spacing > it->max_extra_line_spacing)
 		    it->max_extra_line_spacing = extra_line_spacing;
 		}
@@ -24507,7 +24511,8 @@ maybe_produce_line_number (struct it *it)
      called when no display elements were produced from the
      following line, so the paragraph direction might be unknown.
      Therefore we cheat and add 2 blanks, one on either side.  */
-  pint2str (lnum_buf, it->lnum_width + 1, lnum_to_display);
+  // TRUNG: remove the left margin when displaying line numbers
+  pint2str (lnum_buf, it->lnum_width, lnum_to_display);
   strcat (lnum_buf, " ");
 
   /* Setup for producing the glyphs.  */
@@ -32662,7 +32667,11 @@ gui_produce_glyphs (struct it *it)
 
   if (extra_line_spacing > 0)
     {
-      it->descent += extra_line_spacing;
+      // TRUNG: Vertically centralize extra line spacing
+      int extra_line_spacing_ascent = extra_line_spacing / 2;
+      int extra_line_spacing_descent = extra_line_spacing - extra_line_spacing_ascent;
+      it->ascent += extra_line_spacing_ascent;
+      it->descent += extra_line_spacing_descent;
       if (extra_line_spacing > it->max_extra_line_spacing)
 	it->max_extra_line_spacing = extra_line_spacing;
     }
