diff --git a/src/xdisp.c b/src/xdisp.c
index f6e02cd730..468dbe4807 100644
--- a/src/xdisp.c
+++ b/src/xdisp.c
@@ -24538,7 +24538,7 @@ maybe_produce_line_number (struct it *it)
      called when no display elements were produced from the
      following line, so the paragraph direction might be unknown.
      Therefore we cheat and add 2 blanks, one on either side.  */
-  pint2str (lnum_buf, it->lnum_width + 1, lnum_to_display);
+  pint2str (lnum_buf, it->lnum_width, lnum_to_display);
   strcat (lnum_buf, " ");

   /* Setup for producing the glyphs.  */
